### PR TITLE
Add a lint for unneeded return statements that should be written as exprs

### PR DESCRIPTION
**Pull request by birkenfeld**
I don't know if "misc" is the right file. Let me know if you want this to be in a separate file!

### CODE REVIEW

File: src/misc.rs [hunk 1]
```diff
@@ -239,33 +239,101 @@ impl LintPass for CmpOwned {
...
+                }
+            }
+            _ => { }
+        }
+    }
+
+    fn emit_lint(&mut self, cx: &Context, spans: (Span, Span)) {
+        span_lint(cx, NEEDLESS_RETURN, spans.0, &format!(
+            "unneeded return statement. Consider using {} \
+             without trailing semicolon",
```
Comment (Manishearth) [inline src/misc.rs]: nit: "without the trailing"

**Comment by Manishearth** [comment]
(I'll fix the nit myself, since it's minor)

**Comment by Manishearth** [comment]
Thanks!

**Comment by birkenfeld** [comment]
Thanks! Would you accept further whitespace and indentation cleanup? There is a pretty 50-50 mix between tab and 4-space indentation at the moment...

**Comment by Manishearth** [comment]
Sure! 4-space, no spaces on empty lines is my preferred one (which is in line with Servo and other rust projects)

**Comment by Manishearth** [comment]
I intend to regularly feed clippy to rustfmt, if rustfmt starts working again for me, but whitespace PRs welcome!

**Pull request merged**